GitFyle.Core.Api/Services/Foundations/Configurations/ConfigurationService.Validations.cs

```diff
@@ -0,0 +1,139 @@
+                (Rule: await IsInvalidAsync(configuration.CreatedDate), Parameter: nameof(configuration.CreatedDate)),
+                (Rule: await IsInvalidAsync(configuration.UpdatedDate), Parameter: nameof(configuration.UpdatedDate)),
+
+                (Rule: await IsValuesNotSameAsync(
+                    createdBy: configuration.CreatedBy,
+                    updatedBy: configuration.UpdatedBy,
+                    createdByName: nameof(configuration.CreatedBy)),
+                    Parameter: nameof(configuration.UpdatedBy)),
+
+                (Rule: await IsDatesNotSameAsync(
+                    createdDate: configuration.CreatedDate,
+                    updatedDate: configuration.UpdatedDate,
+                    createdDateName: nameof(configuration.CreatedDate)
+                    ),
+                    Parameter: nameof(configuration.UpdatedDate)),
+
+                (Rule: await IsNotRecentAsync(configuration.CreatedDate), Parameter: nameof(configuration.CreatedDate))
+
+                );
+        }
+
+        private async ValueTask<dynamic> IsNotRecentAsync(DateTimeOffset date)
+        {
+            var (isNotRecent, startDate, endDate) = await IsDateNotRecentAsync(date);
+
+            return new
+            {
+                Condition = isNotRecent,
+                Message = $"Date is not recent. Expected a value between {startDate} and {endDate} but found {date}"
+            };
+        }
+
+        private async ValueTask<(bool IsNotRecent, DateTimeOffset StartDate, DateTimeOffset EndDate)>
+            IsDateNotRecentAsync(DateTimeOffset date)
+        {
+            int pastSeconds = 60;
+            int futureSeconds = 0;
+
+            DateTimeOffset currentDateTime =
+                await this.dateTimeBroker.GetCurrentDateTimeOffsetAsync();
+
+            if (currentDateTime == default)
+            {
+                return (false, default, default);
+            }
+
+            TimeSpan timeDifference = currentDateTime.Subtract(date);
+            DateTimeOffset startDate = currentDateTime.AddSeconds(-pastSeconds);
+            DateTimeOffset endDate = currentDateTime.AddSeconds(futureSeconds);
+            bool isNotRecent = timeDifference.TotalSeconds is > 60 or < 0;
+
+            return (isNotRecent, startDate, endDate);
+        }
+
+        private static async ValueTask<dynamic> IsDatesNotSameAsync(
+            DateTimeOffset createdDate,
+            DateTimeOffset updatedDate,
+            string createdDateName
+            ) => new
+            {
+                Condition = createdDate != updatedDate,
+                Message = $"Date is not same as {createdDateName}"
+            };
+
+        private static async ValueTask<dynamic> IsValuesNotSameAsync(string createdBy, string updatedBy, string createdByName) => new
+        {
+            Condition = createdBy != updatedBy,
+            Message = $"Text is not same as {createdByName}"
+        };
```

Comment:
```suggestion
        private static async ValueTask<dynamic> IsValuesNotSameAsync(
            string createdBy,
            string updatedBy,
            string createdByName) => new
        {
            Condition = createdBy != updatedBy,
            Message = $"Text is not same as {createdByName}"
        };
```
Method signature line over 120 char, rule now lines over 120. Long lines make it hard to read the code on mobile.

Reply:
Done